tools/euclideandistance.py

```diff
@@ -0,0 +1,64 @@
+from math import sqrt
+import numpy as np
+from config import Config
+
+def euclidean_distance(startcoords : list, endcoords : list):
+    xstart = startcoords[0]
+    ystart = startcoords[1]
+    xend = endcoords[0]
+    yend = endcoords[1]
+    distance = sqrt((abs(xend - xstart)) ** 2 + (abs(yend - ystart)) ** 2)
+    return distance
+
+def find_hub_coordinates(region):
+    region_id = region.id
+    xcoord = Config.region_size * 0.5 + (region_id % int(np.sqrt(Config.regions))) * Config.region_size
+    ycoord = Config.region_size * 0.5 + (region_id // int(np.sqrt(Config.regions))) * Config.region_size
+    coords = [xcoord,ycoord]
+    return coords
+
+def find_region(env, coordinates):
+    ids = []
+    for regionid in range(env.config.regions):
+        x1 = env.regions[regionid].geography[0][0]
+        x2 = env.regions[regionid].geography[0][1]
+        y1 = env.regions[regionid].geography[1][0]
+        y2 = env.regions[regionid].geography[1][1]
+
+        if coordinates[0] >= x1 and coordinates[0] <= x2 \
+                and coordinates[1] >= y1 and coordinates[1] <= y2 :
+            ids.append(regionid)
+    region_id = min(ids)
+
+    for region in env.regions:
```

Comment:
No need to make this `O(N)`: when we know the region_id, we know the region: `env.regions[region_id]` (maybe - 1  but that depends on the starting value of the ID. I believe that's 0 so we should be good?).